client-api/src/main/java/org/sonarsource/sonarlint/core/clientapi/client/SuggestBindingParent.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * SonarLint Core - Client API
+ * Copyright (C) 2016-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarsource.sonarlint.core.clientapi.client;
+
+import java.util.List;
+import java.util.Map;
+import org.eclipse.lsp4j.jsonrpc.validation.NonNull;
+import org.sonarsource.sonarlint.core.clientapi.backend.config.binding.BindingSuggestionDto;
+
+public class SuggestBindingParent {
```

Comment:
Even if this will create a bit of duplication, I have a doubt about having a common ancestor here. From an OOP point of view it doesn't make sense in my opinion.
Also we want to keep this client API stable, so we should avoid sharing request/params class between different requests, as a change on one side can lead to an unexpected change on the other side.

Reply:
Good point; Addressed in [078e014](https://github.com/SonarSource/sonarlint-core/pull/594/commits/078e01419b054088432516cdda1ff62c8418bd96)